detox/test/e2e/27.retries.test.js

```diff
@@ -9,6 +12,10 @@ describe('jest.retryTimes() support', () => {
   let counter = 3;
 
   beforeAll(async () => {
+    // This test won't work if you retry it via -R, --retries.
+    // Here we also assert that the session object is accessible from the sandbox.
+    jestExpect(session.testSessionIndex).toBe(0);
```

Comment:
Since this end-to-end test will break **anyway** if it gets retried via `-R, --retries`, I decided to add one more innocent safeguard to make sure that the subject of this pull request doesn't lose test coverage easily.

The core test for that belongs to integration test suite anyway.